docs/docs/guides/developer-guide/channel-aware/index.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Channelaware entities"
+showtoc: true
+---
+
+# Defining channel-aware entities
+
+Making an entity channel-aware means that it can be associated with a specific [Channel](/reference/typescript-api/channel/).
+This is useful when you want to have different data or features for different channels. First you will have to create
+a base entity ([Define a database entity](/guides/developer-guide/database-entity/)) that implements the `ChannelAware` interface.
+This interface requires the entity to have a `channels` property
+
+```ts title="src/plugins/requests/entities/product-request.entity.ts"
+import { DeepPartial } from '@vendure/common/lib/shared-types';
+import { VendureEntity, Product, EntityId, ID, ChannelAware } from '@vendure/core';
+import { Column, Entity, ManyToOne } from 'typeorm';
+
+@Entity()
+class ProductRequest extends VendureEntity {
+    constructor(input?: DeepPartial<ProductRequest>) {
+        super(input);
+    }
+
+    @ManyToOne(type => Product)
+    product: Product;
+
+    @EntityId()
+    productId: ID;
+
+    @Column()
+    text: string;
+
+    @ManyToMany(() => Channel)
```

Comment:
You can make Docusaurus highlight this section with

```
// highlight-start
...
// highlight-end
```

For single-line highlights you can use

```
// highlight-next-line
....
```